oauthutil/oauthutil.go

```diff
@@ -424,6 +424,7 @@ func (s *authServer) Start() {
 			if state != s.state {
 				fs.Debugf(nil, "State did not match: want %q got %q", s.state, state)
 				fmt.Fprintf(w, "<h1>Failure</h1>\n<p>Auth state doesn't match</p>")
+				s.code <- "-1"
```

Comment:
I think returning "" might be better than "-1" here - that wil cause an error to be returned.